### PR TITLE
Improvements image views

### PR DESCRIPTION
**Pull request by Themanwhosmellslikesugar**

### CODE REVIEW

File: images/serializers.py [hunk 1]
```diff
@@ -73,7 +73,27 @@ class Meta:
                   'started_at', 'status_display', 'notes')
 
     def get_flavour_display(self, obj):  # pylint: disable=no-self-use
+        """Returns the flavour for display on client. """
```
Comment (karpa4o4) [inline images/serializers.py]: Move adding docstring to `Rework ImageSerializer` commit.

File: images/serializers.py [hunk 2]
```diff
@@ -73,7 +73,27 @@ class Meta:
...
+
         return obj.get_flavour_display()
 
     def get_status_display(self, obj):  # pylint: disable=no-self-use
+        """Returns the status for display on client. """
```
Comment (karpa4o4) [inline images/serializers.py]: Move adding docstring to `Rework ImageSerializer` commit.

File: images/serializers.py [hunk 3]
```diff
@@ -73,7 +73,27 @@ class Meta:
...
+    image notes. """
+
+    image_id = serializers.UUIDField()
+    notes = serializers.CharField()
+
+    def validate_image_id(self, value):  # pylint: disable=no-self-use
+        """Validates if image with image_id exist. """
+
+        if not Image.objects.filter(image_id=value).exists():
+            raise serializers.ValidationError('Username is already in use.')
```
Comment (karpa4o4) [inline images/serializers.py]: Replace error message to `Image does not exist`.

File: images/views.py [hunk 4]
```diff
@@ -25,3 +30,39 @@ class ListImagesView(generics.ListAPIView):
...
         user = self.request.user
         return Image.objects.filter(user=user)
+
+
+class NotesImageUpdate(generics.UpdateAPIView):
+    """Update the notes of image. """
+
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def put(self, request, *args, **kwargs):
```
Comment (karpa4o4) [inline images/views.py]: You need to check if the image belongs to the current user.

File: images/views.py [hunk 5]
```diff
@@ -25,3 +30,39 @@ class ListImagesView(generics.ListAPIView):
...
+
+        return JsonResponse(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ImageDelete(generics.DestroyAPIView):
+    """Delete the image. """
+
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def delete(self, request, *args, **kwargs):
```
Comment (karpa4o4) [inline images/views.py]: You need to check if the image belongs to the current user.

File: images/views.py [hunk 6]
```diff
@@ -31,6 +32,26 @@ def get_queryset(self):
         return Image.objects.filter(user=user)
 
 
+class ImageDeleteView(generics.DestroyAPIView):
+    """Deletes the image. """
```
Comment (eugulixes) [inline images/views.py]: `"""Deletes an image. """`

File: images/views.py [hunk 7]
```diff
@@ -25,3 +29,24 @@ class ListImagesView(generics.ListAPIView):
     def get_queryset(self):
         user = self.request.user
         return Image.objects.filter(user=user)
+
+
+class ImageNotesUpdateView(generics.UpdateAPIView):
+    """Update the notes of the image. """
```
Comment (eugulixes) [inline images/views.py]: `"""Updates image notes. """`

File: images/serializers.py [hunk 8]
```diff
@@ -83,8 +83,32 @@ def get_status(self, obj):  # pylint: disable=no-self-use
...
+
+        if image.user != self.user:
+            raise serializers.ValidationError('Current user does not have current image.')
+
+        return value
+
+
 class ImageNotesUpdateSerializer(serializers.Serializer):  # pylint: disable=abstract-method
-    """Serializes both the image_id and the notes provided by the current user to update their
+    """Serializes both the image_id and notes provided by the current user to update their
```
Comment (eugulixes) [inline images/serializers.py]: Why did you remove `the`?

**Review by eugulixes — APPROVED** [review]
